src/SpreadingDye.jl

```diff
@@ -33,74 +33,103 @@ function random_point_on_domain(m::AbstractMatrix{Bool})
     x
 end
 
-# grow the range one at a time
-function grow!(georange::AbstractMatrix{Bool}, edges::Set, dom::AbstractMatrix{Bool}, algo::Symbol; ignore_domain = false)
-    newcell = isempty(edges) ? random_point_on_domain(georange) : first(rand(edges))
-    while georange[newcell...]
-        isempty(edges) && push!(edges, jump(georange, dom, algo)) # allows for patchy ranges
-        edge, newcell = rand(edges)
-        pop!(edges, (edge, newcell))
+# choose a random new cell
+function _random_edge_or_jump(georange, edges, dom, algo)
+    while !isempty(edges)
+        newidx = rand(eachindex(edges))
+        newcell = popat!(edges, newidx)
+        !georange[newcell...] && return newcell
     end
-    georange[newcell...] = true
-    for nb in algos[algo]
+    return jump(georange, dom, algo)
+end
+
+# grow the range one at a time
+grow!(georange, edges, dom, algo::Symbol; ignore_domain = false) =
+    _grow!(georange, edges, dom, algos[algo]; ignore_domain)
+
+function _grow!(georange::AbstractMatrix{Bool}, edges::Vector, dom::AbstractMatrix{Bool}, algo::Tuple; ignore_domain = false)
+    newcell = _random_edge_or_jump(georange, edges, dom, algo)
+    for nb in algo
         neighbor = newcell .+ nb
-        if within_edges(dom, neighbor) && (ignore_domain || dom[neighbor...]) && !georange[neighbor...]
```

Comment:
Am I correct that this is a stylistic change to the logical expression? That is unrelated and causes some confusion, so should definitely be a separate PR.